packages/flutter_tools/lib/src/android/android_sdk.dart

```diff
@@ -409,21 +409,15 @@ class AndroidSdk {
     return null;
   }
 
-  /// First try Java bundled with Android Studio, then sniff JAVA_HOME, then fallback to PATH.
-  static String? findJavaBinary({
+  static String? findJavaHome({
```

Comment:
Factoring this out makes a lot of sense. It should get some dartdoc, though — particularly as its name sounds like it might just look at JAVA_HOME, but it actually does a couple more things.

```suggestion
  /// A value that would be appropriate to use as JAVA_HOME.
  ///
  /// This method considers, in order:
  /// * the JDK bundled with Android Studio, if one is found;
  /// * the JAVA_HOME in the ambient environment, if set;
  /// * on macOS, the value provided by the `java_home` command.
  static String? findJavaHome({
```